pkg/kcp/provider/aws/vpcpeering/kcpNetworkLocalLoad.go

```diff
@@ -0,0 +1,70 @@
+package vpcpeering
+
+import (
+	"context"
+	"fmt"
+	cloudcontrolv1beta1 "github.com/kyma-project/cloud-manager/api/cloud-control/v1beta1"
+	"github.com/kyma-project/cloud-manager/pkg/composed"
+	"github.com/kyma-project/cloud-manager/pkg/util"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func kcpNetworkLocalLoad(ctx context.Context, st composed.State) (error, context.Context) {
+	state := st.(*State)
+	logger := composed.LoggerFromCtx(ctx)
+
+	net := &cloudcontrolv1beta1.Network{}
+
+	namespace := state.ObjAsVpcPeering().Namespace
+
+	logger = logger.
+		WithValues("localKcpNetwork", fmt.Sprintf("%s/%s", namespace, state.ObjAsVpcPeering().Spec.Details.LocalNetwork.Name))
+
+	err := state.Cluster().K8sClient().Get(ctx, client.ObjectKey{
+		Namespace: namespace,
+		Name:      state.ObjAsVpcPeering().Spec.Details.LocalNetwork.Name,
+	}, net)
+
+	if client.IgnoreNotFound(err) != nil {
+		return composed.LogErrorAndReturn(err, "Error loading local KCP Network for KCP VpcPeering", composed.StopWithRequeueDelay(util.Timing.T10000ms()), ctx)
+	}
+
+	if apierrors.IsNotFound(err) {
+		state.ObjAsVpcPeering().Status.State = string(cloudcontrolv1beta1.ErrorState)
+		return composed.PatchStatus(state.ObjAsVpcPeering()).
+			SetExclusiveConditions(metav1.Condition{
+				Type:    cloudcontrolv1beta1.ConditionTypeError,
+				Status:  metav1.ConditionTrue,
+				Reason:  cloudcontrolv1beta1.ReasonMissingDependency,
+				Message: "Local network not found",
+			}).
+			ErrorLogMessage("Error patching KCP VpcPeering status with missing local network dependency").
+			SuccessError(composed.StopWithRequeueDelay(util.Timing.T10000ms())).
+			SuccessLogMsg("KCP VpcPeering local KCP Network not found").
+			Run(ctx, state)
+	}
+
+	if net.Status.State != string(cloudcontrolv1beta1.ReadyState) {
```

Comment:
Services MUST NOT depend on the status state which is arbitrary natural language communication to the user. It MUST use conditions.